Feat/chat deprecations (#23)
mixer/developer-docs: pages/03.content/02.blog/03.ChatListDeprecations/blog-item.md

---
title: Deprecations coming to Chat Viewer List
date: 14:00 11/05/2018
author: rifox
taxonomy:
    category: blog
    tag: [blog, sdk, client, deprecations]
---

> As Mixer grows, our API needs to grow too this sometimes leads to deprecations. With the power of our new developer site, we're taking a more active role in announcing them. You'll see a blog post from our team detailing the changes we're making and what you need to do. The first example of this is with the deprecation of V1 of the Chat Viewer List API.

Reviewer: ```suggestion
As Mixer grows, our API needs to grow too. This sometimes leads to deprecations. With the power of our new developer site, we're taking a more active role in announcing them. You'll see a blog post from our team detailing the changes we're making and what you need to do. The first example of this is with the deprecation of V1 of the Chat Viewer List API.
```

# Chat Viewer List API

With the introduction to V2 over, let's move onto the changes.

> A new version of the Chat viewer List API is now available. It exists in the V2 path.

Reviewer: ```suggestion
A new version of the Chatters List API is now available. It exists in the V2 path.
```

Reviewer 2: good catch Jamy

It has the following changes:
- Removes the `/chats/{channelId}/friends` endpoint. This endpoint is not being used in any Mixer Application and we've seen a very small usage of it.
- Changes the calling pattern of `/chats/{channelId}/users.` This will affect pagination.
- Removing the `/chats/{channelId}/users/search` endpoint. Instead supply a `username` query parameter to `/chats/{channelId}/users` to search.

> # Deprecating Chat Viewer List API V1

Reviewer: ```suggestion
# Deprecating Chatters List API V1
```

> With V2 available we plan to deprecate the chat viewer list v1 API on 12/08/2018.

Reviewer: ```suggestion
With V2 available we plan to deprecate the chatters list v1 API on 12/08/2018.
```

## What do I need to do?

> Before 12/08/2018, if you use the V1 Chat Viewer List API. Move to V2.

Reviewer: ```suggestion
Before 12/08/2018, if you use the V1 Chatters List API. Move to V2.
```
